fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/FedoraEventImpl.java

```diff
@@ -58,56 +73,57 @@
+
+    private final static Logger LOGGER = getLogger(FedoraEventImpl.class);
+
     private final String path;
     private final String userID;
-    private final String userData;
-    private final long date;
+    private final Instant date;
     private final Map<String, String> info;
     private final String eventID;
+    private final Set<String> eventResourceTypes;
 
     private final Set<EventType> eventTypes = new HashSet<>();
-    private final Set<String> eventProperties = new HashSet<>();
 
     private static final List<Integer> PROPERTY_TYPES = asList(Event.PROPERTY_ADDED,
             Event.PROPERTY_CHANGED, Event.PROPERTY_REMOVED);
 
-    private static final UniqueValueSupplier pidMinter = new DefaultPathMinter();
-
     /**
      * Create a new FedoraEvent
      * @param type the Fedora EventType
      * @param path the node path corresponding to this event
+     * @param resourceTypes the rdf types of the corresponding resource
      * @param userID the acting user for this event
-     * @param userData any user data for this event
      * @param date the timestamp for this event
      * @param info supplementary information
      */
-    public FedoraEventImpl(final EventType type, final String path, final String userID,
-            final String userData, final long date, final Map<String, String> info) {
-        this(singleton(type), path, userID, userData, date, info);
+    public FedoraEventImpl(final EventType type, final String path, final Set<String> resourceTypes,
+            final String userID, final Instant date, final Map<String, String> info) {
+        this(singleton(type), path, resourceTypes, userID, date, info);
     }
 
    /**
      * Create a new FedoraEvent
      * @param types a collection of Fedora EventTypes
      * @param path the node path corresponding to this event
+     * @param resourceTypes the rdf types of the corresponding resource
      * @param userID the acting user for this event
-     * @param userData any user data for this event
      * @param date the timestamp for this event
      * @param info supplementary information
      */
-    public FedoraEventImpl(final Collection<EventType> types, final String path, final String userID,
-            final String userData, final long date, final Map<String, String> info) {
+    public FedoraEventImpl(final Collection<EventType> types, final String path, final Set<String> resourceTypes,
+            final String userID, final Instant date, final Map<String, String> info) {
         requireNonNull(types, "FedoraEvent requires a non-null event type");
         requireNonNull(path, "FedoraEvent requires a non-null path");
 
         this.eventTypes.addAll(types);
         this.path = path;
+        this.eventResourceTypes = resourceTypes;
         this.userID = userID;
-        this.userData = userData;
         this.date = date;
         this.info = isNull(info) ? emptyMap() : info;
-        this.eventID = pidMinter.get();
```

Comment:
Sounds good. We should update the `InternalAuditor` to create the "path-minter-like" hierarchy as a part of this effort... as well as making fcrepo-audit compile ;)

Additionally, similar logic will need to be implemented for the "external auditor" found in the fcrepo-camel-toolbox, since it uses the eventID as the subject of the audit triples stored in the external triplestore. It is important that the external triplestore audit subjects are the same URIs that are found in the repository:
https://github.com/fcrepo4-exts/fcrepo-camel-toolbox/blob/master/fcrepo-audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java#L84